src/hooks/useUpdateEffect.ts

```diff
@@ -0,0 +1,14 @@
+import { useRef, useEffect } from 'react'
```

Comment:
could you explain what this hook is doing. I'm not sure I understand its intent.

Reply:
https://reactjs.org/docs/hooks-reference.html#useref
The only purpose of using `useRef` here is to store the boolean value.

Reply:
I have added `useUpdateEffectHook` so that the hook is called only after any of the dependency changes and skip during initial render.
Use case: In initial loading of patient info, `searchPatients` should be called only once.
Later,`searchPatients` should be called when `userPageRequest` changes.

Reply:
I thought of adding this hook because adding just `userPageRequest ` in dependency array  didn't work here https://github.com/HospitalRun/hospitalrun-frontend/blob/6d15f9fe91af0407066bba430e70d770a1a2fe96/src/patients/list/ViewPatients.tsx#L85

So, thought of segregating `debouncedSearchText` and `userPageRequest` on change callbacks.